### ISSUE TITLE
Coverage annotations publication failed

### ISSUE THREAD

**Issue by astb01** [ArtiomTr/jest-coverage-report-action#327]
### Describe a bug

When I use the action it seems to fail to submit the annotations

Below is my workflow snippet:

```
name: 'Development Workflow'

on:
  push:
    branches:
      - develop

permissions:
  contents: read

jobs:
  build:
    name: Linting and Test
    runs-on: ubuntu-latest
    strategy:
      matrix:
        node-version: [17.x]
    steps:
      - name: Perform Linting
        uses: actions/checkout@v3
      - name: Use Node.js ${{ matrix.node-version }}
        uses: actions/setup-node@v2
        with:
          node-version: ${{ matrix.node-version }}
          cache: 'npm'
      - run: npm install
      - run: npx eslint --ext .js,.jsx src/ --fix-dry-run
      - name: Run Tests
        uses: actions/checkout@v3
      - name: Use Node.js ${{ matrix.node-version }}
        uses: actions/setup-node@v2
        with:
          node-version: ${{ matrix.node-version }}
          cache: 'npm'
      - run: npm install
      - run: npx jest

  coverage:
    needs: build
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v1
      - uses: artiomtr/jest-coverage-report-action@v2
        id: coverage
        with:
          github-token: ${{ secrets.GITHUB_TOKEN }}
          threshold: 80
          output: report-markdown
      - uses: marocchino/sticky-pull-request-comment@v2
        with:
          # pass output from the previous step by id.
          message: ${{ steps.coverage.outputs.report }}
```

I'm seeing:

```
Begin coverage annotations publication...
Coverage annotations publication failed
Error: HttpError: Resource not accessible by integration
HttpError: Resource not accessible by integration
    at /home/runner/work/_actions/artiomtr/jest-coverage-report-action/v2/dist/index.js:8:323830
    at processTicksAndRejections (node:internal/process/task_queues:[96](https://github.com/astb01/places-app/actions/runs/3441883547/jobs/5741909193#step:3:98):5)
    at async /home/runner/work/_actions/artiomtr/jest-coverage-report-action/v2/dist/index.js:2230:15[97](https://github.com/astb01/places-app/actions/runs/3441883547/jobs/5741909193#step:3:99)7
    at async ci (/home/runner/work/_actions/artiomtr/jest-coverage-report-action/v2/dist/index.js:2221:158)
    at async jD (/home/runner/work/_actions/artiomtr/jest-coverage-report-action/v2/dist/index.js:2230:15736)
Coverage annotations publication ended
Error: Jest coverage report action failed
Error: Process completed with exit code 1.
```

### Expected behavior

Expect the coverage report to be published and available in my PR

### Details

-   Action version: v2
-   OS, where your action is running (windows, linux): ubuntu-latest

-   <details><summary>Screenshots</summary>

        

        

    </details>

**Comment by imilchev** [comment]
same here

**Comment by LeonardodeLima** [comment]
same here in version v2.1.2

**Comment by jlamb1** [comment]
add the permissions listed in this section: https://github.com/ArtiomTr/jest-coverage-report-action#pull-request-number
```
  checks: write
  pull-requests: write
  contents: write
  ```

**Comment by ArtiomTr** [comment]
Hey @astb01, @imilchev, @LeonardodeLima and @jlamb1 :wave:,

Sorry for the late reply. This issue is fixed in the `v2.2.0` (or just `v2`) version. [The documentation on how to properly configure action on forks with no write permissions is here](https://github.com/ArtiomTr/jest-coverage-report-action#forks-with-no-write-permission).

**Comment by sam3k** [comment]
I'm getting a similar error but I am using a PAT. 

![image](https://github.com/ArtiomTr/jest-coverage-report-action/assets/180152/ff9f778e-09da-435a-998f-e55ebc8462a4)

Any ideas?
